File: DSA/6.0_Linked_List/boiler_plate.py
# Helper code to run leetcode solutions locally
from typing import List, Optional
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def array_to_list(arr):
    if not arr:
        return None
    head = ListNode(arr[0])
    current = head
    for i in range(1, len(arr)):
        current.next = ListNode(arr[i])
        current = current.next
    return head

def print_list(node):
    while node:
        print(node.val, end=" -> " if node.next else "\n")
        node = node.next

File: DSA/LinkedLists/boiler_plate.js
// Helper code to run leetcode solutions locally
function ListNode(val, next) {
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? null : next)
}

function arrayToList(arr) {
    if (arr.length === 0) return null;
    let head = new ListNode(arr[0]);
    let current = head;
    for (let i = 1; i < arr.length; i++) {
        current.next = new ListNode(arr[i]);
        current = current.next;
    }
    return head;
}

// we can run these leetcode problems locally by using these helper functions as a chrome snippet